src/bin/pgcopydb/schema.c

```diff
@@ -848,13 +848,171 @@ struct FilteringQueries listSourceTableSizeSQL[] = {
 
 
 /*
- * schema_prepare_pgcopydb_table_size creates a table named pgcopydb_table_size
- * on the given connection (typically, the source database). The creation is
- * skipped if the table already exists.
+ * For code simplicity the index array is also the SourceFilterType enum value.
+ */
+struct FilteringQueries listSourceTableSizeEstimateSQL[] = {
+	{
+		SOURCE_FILTER_TYPE_NONE,
+
+		"  select c.oid, c.relpages * 8192 as bytes, "
+		"         pg_size_pretty(c.relpages * 8192) "
```

Comment:
We already have c.relpages and c.reltuples::bigint in the `listSourceTablesSQL`, I think we should re-use them rather than fetch them again from the source database. Maybe we can use a SQLite query and still populate the `s_table_size` table, which seems like a good idea to simplify our work later. If it's not possible to have SQLite compute `pg_size_pretty()` then we could compute that on the fly when reading from the SQLite catalogs and the `bytes_pretty` column is NULL when the `bytes` column is populated.

Now that we have our own SQLite copy vendored-in, I wonder how difficult it is to enhance it with our own SQL functions, that said. We might want to implement `pg_size_pretty` as a SQLite function. If that proves easy enough, later we might even implement an LSN type in SQLite.

Reply:
I updated `getTableArray` function to also record the table size estimates for each table. This function was responsible for looping through the result of `listSourceTablesSQL` that is mentioned on your comment.

I did not need to implement `pg_size_pretty` as I could reuse `pretty_print_bytes` function that has a similar purpose.

Reply:
Ah, that's a nice and simple approach to it, cheers.